cmd/helm/install.go

```diff
@@ -240,6 +245,20 @@ func isChartInstallable(ch *chart.Chart) (bool, error) {
 	return false, errors.Errorf("%s charts are not installable", ch.Metadata.Type)
 }
 
+// It is to limit the total size of the template file to 3m, and the content of files over 3m should not be allowed to be installed.
+//
+// This is because the server requires the requestbody to be less than 3m when creating secrets.For details, please refer to: https://github.com/kubernetes/kubernetes/blob/release-1.18/staging/src/k8s.io/apiserver/pkg/server/config.go#L323
+func isLimitSize(raw []*chart.File) (bool, error) {
+	totalSize := int64(0)
+	for _, r := range raw {
+		totalSize += int64(bytes.Count(r.Data, []byte("")))
+	}
+	if totalSize < int64(3 * 1024 * 1024) {
+		return true, nil
+	}
```

Comment:
3mb is an issue for secrets and only when using etcd as the backing storage for those. Helm can use a database and that has different size constraints. Hard coding one size may not be the best way to check for this.

Reply:
Hard coding one size the worst-case scenario
.And I submitted an update and look forward to your suggestion.